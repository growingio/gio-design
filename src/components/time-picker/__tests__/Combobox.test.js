import React from 'react';
import { act } from 'react-dom/test-utils';
import { waitFor } from '@testing-library/react';
import { mount, render } from 'enzyme';
import { Moment } from 'moment';
import Combobox from '../Combobox';

// 打印快照
describe('Testing Combobox', () => {
  it('should match alert base snapshot.', () => {
    const wrapper = render(<Combobox />);
    expect(wrapper).toMatchSnapshot();
  });
  //  props test
  it('props test', () => {
    const onClickMock = jest.fn();
    expect(
      mount(
        <Combobox
          defaultOpenValue={false}
          use12Hours
          isAM
          onAmPmChange={onClickMock}
          //   onChange={onClickMock}
          value={Moment}
        />
      ).exists('.gio-time-picker-panel-combobox')
    ).toMatchSnapshot();
  });

  it('props test2', () => {
    const onClickMock = jest.fn();
    expect(
      mount(
        <Combobox
          defaultOpenValue={false}
          use12Hours={false}
          isAM
          onAmPmChange={onClickMock}
        />
      ).exists('.gio-time-picker-panel-combobox')
    ).toMatchSnapshot();
  });
});
