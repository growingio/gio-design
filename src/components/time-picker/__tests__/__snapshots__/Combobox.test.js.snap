// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Combobox props test 1`] = `false`;

exports[`Testing Combobox props test2 1`] = `false`;

exports[`Testing Combobox should match alert base snapshot. 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "undefined-combobox",
    },
    "children": Array [],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
